Assets/MixedRealityToolkit/Services/BaseDataProviderAccessCoreSystem.cs

```diff
@@ -45,21 +78,73 @@ public override void Update()
         {
             base.Update();
 
-            foreach (var provider in dataProviders)
+            // Applications may choose to trigger service changes (i.e. setting MixedRealityToolkit.ActiveProfile) in
+            // response to user input, such as a button press. Since most code runs on Unity's main thread, the collection
+            // of data providers may change mid-Update, causing an InvalidOperationException to be thrown during enumeration.
+            // To avoid unneccessary error logging, we catch the exception. It is safe to abort the loop in this fashion since
+            // it is extremely unlikely that the data providers being enumerated are the same as when we started enumeration.
+            try
             {
-                provider.Update();
+                foreach (IMixedRealityDataProvider provider in dataProviders)
+                {
+                    try
+                    {
+                        provider.Update();
+                    }
+                    catch (Exception e)
+                    {
+                        Debug.LogError($"{provider.Name}'s Update method threw {e.GetType()}");
```

Comment:
Would be good to mention that the exception was ignored because we catch and continue here.

Reply:
added "Continuing to the next service" to each of these messages.